src/Request.ts

```diff
@@ -178,3 +178,46 @@ export function isRetryable (status?: number): boolean {
       ].indexOf(status) !== -1)
     )
 }
+
+export function fetch(endpoint: string): Promise<string> {
+  return new Promise<string>((resolve, reject) => {
+    const parsedUrl = url.parse(endpoint)
+
+    const req = (parsedUrl.protocol === 'https:' ? https : http).get(endpoint, res => {
+      res.pipe(concat((bodyBuffer: Buffer) => {
+        if (res.statusCode === 200) {
+          return resolve(bodyBuffer.toString())
+        }
+
+        const err = new UploadError(`HTTP status ${res.statusCode} received from bundle server`)
```

Comment:
`UploadError` probably wants a more generic name like `RequestError` or `NetworkError` now

Reply:
If we're not using the retry logic then I think it can be a plain old `Error`

Reply:
Renamed to `NetworkError`